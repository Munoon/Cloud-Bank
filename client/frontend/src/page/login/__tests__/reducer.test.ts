import reducer, {
    loginErrorAction,
    loginLoadingAction,
    loginLoggedInAction,
    loginShowPassword,
    LoginState,
    LoginStoreStateType
} from '../reducer';

describe('login store reducer test', () => {
    describe('functional tests', () => {
        it('loading', () => {
            const expected: LoginStoreStateType = {
                state: LoginState.LOADING,
                showPassword: false,
                errorData: null
            };

            const store: LoginStoreStateType = {
                state: LoginState.IDLE,
                showPassword: false,
                errorData: { errorCode: 'incorrect_password' }
            };

            const actual = reducer(store, loginLoadingAction());
            expect(actual).toEqual(expected);
        });

        it('logged in', () => {
            const expected: LoginStoreStateType = {
                state: LoginState.LOGGED_IN,
                showPassword: false,
                errorData: null
            };

            const store: LoginStoreStateType = {
                state: LoginState.IDLE,
                showPassword: false,
                errorData: { errorCode: 'incorrect_password' }
            };

            const actual = reducer(store, loginLoggedInAction());
            expect(actual).toEqual(expected);
        });

        it('login error', () => {
            const errorData = { errorCode: 'incorrect_password', errorMessage: 'Incorrect password' };
            const expected: LoginStoreStateType = {
                state: LoginState.ERROR,
                showPassword: false,
                errorData: errorData
            };

            const actual = reducer(undefined, loginErrorAction(errorData));
            expect(actual).toEqual(expected);
        });

        it('password toggle on', () => {
            const expected: LoginStoreStateType = {
                state: LoginState.IDLE,
                showPassword: true,
                errorData: null
            };

            const actual = reducer(undefined, loginShowPassword(true));
            expect(actual).toEqual(expected);
        });

        it('password toggle off', () => {
            const store: LoginStoreStateType = {
                state: LoginState.IDLE,
                showPassword: true,
                errorData: null
            };

            const expected: LoginStoreStateType = {
                state: LoginState.IDLE,
                showPassword: false,
                errorData: null
            };

            const actual = reducer(store, loginShowPassword(false));
            expect(actual).toEqual(expected);
        });
    });

    describe('snapshot tests', () => {
        it('loading', () => {
            const store: LoginStoreStateType = {
                state: LoginState.IDLE,
                showPassword: false,
                errorData: { errorCode: 'incorrect_password' }
            };

            const actual = reducer(store, loginLoadingAction());
            expect(actual).toMatchSnapshot();
        });

        it('logged in', () => {
            const store: LoginStoreStateType = {
                state: LoginState.IDLE,
                showPassword: false,
                errorData: { errorCode: 'incorrect_password' }
            };

            const actual = reducer(store, loginLoggedInAction());
            expect(actual).toMatchSnapshot();
        });

        it('login error', () => {
            const errorData = { errorCode: 'incorrect_password', errorMessage: 'Incorrect password' };
            const actual = reducer(undefined, loginErrorAction(errorData));
            expect(actual).toMatchSnapshot();
        });

        it('password toggle', () => {
            const actual1 = reducer(undefined, loginShowPassword(true));
            expect(actual1).toMatchSnapshot();

            const store: LoginStoreStateType = {
                state: LoginState.IDLE,
                showPassword: true,
                errorData: null
            };

            const actual2 = reducer(store, loginShowPassword(false));
            expect(actual2).toMatchSnapshot();
        });
    });
});