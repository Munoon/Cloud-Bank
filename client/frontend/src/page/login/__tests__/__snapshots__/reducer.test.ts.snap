// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login store reducer test snapshot tests loading 1`] = `
Object {
  "errorData": null,
  "showPassword": false,
  "state": "login_state:loading",
}
`;

exports[`login store reducer test snapshot tests logged in 1`] = `
Object {
  "errorData": null,
  "showPassword": false,
  "state": "login_state:logged_in",
}
`;

exports[`login store reducer test snapshot tests login error 1`] = `
Object {
  "errorData": Object {
    "errorCode": "incorrect_password",
    "errorMessage": "Incorrect password",
  },
  "showPassword": false,
  "state": "login_state:error",
}
`;

exports[`login store reducer test snapshot tests password toggle 1`] = `
Object {
  "errorData": null,
  "showPassword": true,
  "state": "login_state:idle",
}
`;

exports[`login store reducer test snapshot tests password toggle 2`] = `
Object {
  "errorData": null,
  "showPassword": false,
  "state": "login_state:idle",
}
`;
